server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/DownloadStepdefs.java

```diff
@@ -0,0 +1,102 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.methods.integration.cucumber;
+
+import static com.jayway.restassured.RestAssured.with;
+
+import javax.inject.Inject;
+
+import com.jayway.restassured.http.ContentType;
+import com.jayway.restassured.response.Response;
+
+import cucumber.api.java.en.Given;
+import cucumber.api.java.en.Then;
+import cucumber.api.java.en.When;
+import cucumber.runtime.java.guice.ScenarioScoped;
+
+@ScenarioScoped
+public class DownloadStepdefs {
+
+    private final UserStepdefs userStepdefs;
+    private final MainStepdefs mainStepdefs;
+    private Response response;
+
+    @Inject
+    private DownloadStepdefs(MainStepdefs mainStepdefs, UserStepdefs userStepdefs) {
+        this.mainStepdefs = mainStepdefs;
+        this.userStepdefs = userStepdefs;
+    }
+
+    @Given("^an unknown user with username \"([^\"]*)\" and password \"([^\"]*)\"$")
+    public void createUserWithPassword(String username, String password) throws Exception {
+        mainStepdefs.jmapServer.serverProbe().addUser(username, password);
+    }
+
+    @When("^asking the download endpoint$")
+    public void getDownload() throws Exception {
+        if (userStepdefs.accessToken != null) {
+            response = getDownloadWithAuthorization();
+        } else {
+            response = getDownloadWithoutAuthorization();
+        }
+    }
+
+    private Response getDownloadWithoutAuthorization() {
```

Comment:
you failed at composing things : the two methods share 3  lines, why don't you init the builder this the common stuff then put the authorization header or not depending on accessToken availability ?